fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/Storage.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.storage;
+
+import net.fabricmc.fabric.api.lookup.v1.block.BlockApiLookupRegistry;
+import net.fabricmc.fabric.api.lookup.v1.item.ItemApiLookupRegistry;
+import net.fabricmc.fabric.api.transfer.v1.transaction.Transaction;
+import net.fabricmc.fabric.impl.transfer.TransferApiImpl;
+
+/**
+ * An object that can store resources.
+ *
+ * <p><b>Important note:</b> Unless otherwise specified, all transfer functions take a non-empty resource
+ * and a non-negative maximum amount as parameters.
+ * Implementations are encouraged to throw an exception if these preconditions are violated.
+ *
+ * <p>For transfer functions, the returned amount must be non-negative, and smaller than the passed maximum amount.
+ * Consumers of these functions are encourage to throw an exception if these postconditions are violated.
+ * @param <T> The type of the stored resources.
```

Comment:
I would give some examples of what the stored resources can be. should i=I store `Item`s or `ItemStack`s?

Reply:
`ItemKey`

Reply:
It should be obvious when you see how it's defined in ItemApi, although I agree that this needs more code examples.